### ISSUE TITLE
DateTimeFormat with pattern prints `12:59 AM` with Europe timezone

### ISSUE THREAD

**Issue by potmo** [JodaOrg/joda-time#268]
When using `DateTimeFormat` and `print`ing a pattern with Stockholm Timezone it will 59 minutes past midnight instead of 59 minutes into a new day.

```
package com.potmo.jodabug;

import org.joda.time.DateTimeZone;
import org.joda.time.format.DateTimeFormat;
import org.junit.Test;

import java.util.Locale;

import static org.hamcrest.CoreMatchers.equalTo;
import static org.hamcrest.MatcherAssert.assertThat;

public class JodaTimeZoneErrorTest {

  // Thu, 31 Dec 2015 23:59:59 GMT (unix epoch, seconds)
  private static final long LAST_SECOND_OF_YEAR_EPOCH = 1451606399L;

  @Test
  public void testYearPassingWithUSFormatting() throws Exception {
    DateTimeZone STOCKHOLM = DateTimeZone.forID("Europe/Stockholm");
    String result = DateTimeFormat.forPattern("MM/dd/yy h:mm a")
                                  .withLocale(Locale.US)
                                  .withZone(STOCKHOLM)
                                  .print(LAST_SECOND_OF_YEAR_EPOCH * 1000);

    assertThat(result, equalTo("01/01/16 00:59 AM")); // wrong actual value: 01/01/16 12:59 AM
  }

}
```

**Comment by potmo** [comment]
Sorry. My bad. This is not a bug. Its a feature. 
I just spoke to an american an this is actually the way it should be. The american unit-system just became even stranger.

**Comment by johnjaylward** [comment]
Nisse, If it was 24-hour format then you would be correct to expect the
00:59, but when using 12-hour format, midnight is 12:00 AM. This is due to
round wall clocks not having a 0, but instead the numbers 1-12. Since
midnight (00:00) is the new day, we us AM for that first 12, and PM for the
noon 12. Hope that helps your understanding of where the units came from. I
remember thinking it was weird in grade school myself since it was still
night, shouldn't the midnight 12 be PM. But alas, it's not.

On Fri, Apr 24, 2015 at 10:18 AM, Nisse Bergman <notifications@github.com>
wrote:

> Closed #268 <https://github.com/JodaOrg/joda-time/issues/268>.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JodaOrg/joda-time/issues/268#event-289326630>.
>



-- 


John.
